docs/COREDUMP.md

```diff
@@ -0,0 +1,148 @@
+means the coredump handler runs for a very short time only, and the potentially
+*heavy* and security sensitive coredump processing work is done as part of the
+specified service unit, and thus can take benefit of regular service resource
+management and sandboxing.
+
+The `systemd-coredump` handler will extract a backtrace from any coredumps it
+receieves, which it will then log to the journal. Coredumps thus automatically
+become a loggable event. Further analysis of coredumps is also available via
+the
+[`coredumpctl`](https://www.freedesktop.org/software/systemd/man/coredumpctl.html)
+tool, for example for directly invoking a debugger such as `gdb` on a collected
+coredump.
+
+The handler will write any coredump it collects to
+`/var/lib/systemd/coredump/`, where it will eventually be removed via
+[`tmpfiles.d/`](https://www.freedesktop.org/software/systemd/man/tmpfiles.d.html)
+file "aging".
+
+## User Experience
+
+With the above, any coredumps generated on the system are by default collected
+and turned into logged events — except during very early boot and late
+shutdown. Individual services, processes or users can opt-out of coredump
+collection, by setting `LIMIT_CORE` to 0 (or alternatively invoke
+[`PR_SET_DUMPABLE`](https://man7.org/linux/man-pages/man2/prctl.2.html)). The
+resource limit can be set freely by daemons/processes/users to arbitrary
+values, which the coredump handler will respect. The `coredumpctl` tool may be
+used to further analyze/debug coredumps.
+
+## Alternative Coredump Handlers
+
+While we recommend usage of the `systemd-coredump` handler, it's fully
+supported to use alternative coredump handlers in its palce. For that it is
+recommended to follow similar implementation patterns. Specifically:
+
+1. Use a `sysctl.d/` drop-in to register your handler with the kernel. Make
+   sure to include the `%c` specifier in the pattern (which reflects the
+   crashing process' `RLIMIT_CORE`) and act on it: truncate receieved coredumps
+   once the limit is reached.
+
+2. Do not do heavy processing directly in the coredump handler. Instead,
+   quickly pass off the kernel's coredump serialization file descriptor to an
+   auxiliary service running as service under the service manager, so that it
+   can be done under supervision/resource management, and so on.
+
+Note that at any given time only a single handler can be enabled, i.e. the
+`core_pattern` cannot reference multiple executables.
+
+## Packaging
+
+It might make sense to split `systemd-coredump` into a separate distribution
+package. If doing so, make sure that `/usr/lib/sysctl.d/50-coredump.conf` and
+the associated service and socket units are also added to the split off package.
+
+Note that in a scenario where `systemd-coredump` is split out and not
+installed, coredumping is turned off during the entire runtime of the system —
+unless an alternative handler is installed, or behaviour is manually reverted
+to 1990's style handling (see below).
```

Comment:
The 1990s were great. We had Power Rangers, Limp Bizkit and existential dread due to the first visible signs of irreversible climate collapse. Anyway, I'd suggest simply saying `legacy style handling` instead.